feature(Lab1Model, ConsoleLoader): первая бизнес-логика проекта

TemZZZ opened the pull request:
- добавлен интерфейс IComponent (IComponent.cs)
- добавлен абстрактный класс AbstractComponent (AbstractComponent.cs)
- реализованы классы резистора, катушки, конденсатора (PassiveComponents.cs)
- добавлено консольное приложение ConsoleLoader

iKalentii commented on ConsoleLoader/Program.cs:
```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using PassiveComponents;
+
+
```
- в коде необходимо придерживаться одного стандарта кодирования. Мы используем соглашение RSDN https://rsdn.org/article/mag/200401/codestyle.XML, в планах есть переезд на более прогрессивный стандарт с автоматизацией проверки.

iKalentii commented on ConsoleLoader/Program.cs:
```diff
@@ -0,0 +1,80 @@
+using PassiveComponents;
+
+
+public static class Program
```
- Для всех классов и их членов необходимо ставить XML комментарии для пояснения их работы. XML комментарии ставятся автоматически, если написать три символа «/» возле класса/члена класса. Далее необходимо написать – что это за класс/член класса и для чего используется. Такой способ документирования позволяет при наведении компьютерной мыши на переменную получать всю необходимую информацию. Подробнее про XML комментарии можно почитать тут: https://docs.microsoft.com/ru-ru/dotnet/csharp/programming-guide/xmldoc/xml-documentation-comments

iKalentii commented on Lab1Model/PassiveComponents.cs:
```diff
@@ -0,0 +1,65 @@
+﻿using System;
+
+
+namespace PassiveComponents
```
Следуй рекомендации - одна сущность (класс, интерфейс, структура, перечисление и пр.) - один .cs файл. Это делается для удобства просмотра структуры проекта в обозревателе.

iKalentii commented on Lab1Model/PassiveComponents.cs:
```diff
@@ -0,0 +1,65 @@
+	public class Resistor : Component
+	{
+		public Resistor() { Value = 0; }
+		public Resistor(double value) { Value = value; }
```
Тут можно использовать цепочку конструкторов, чтобы уменьшить дублирование.
`public Resistor(): this(0) {}`

Такая конструкция вызовет конструктор с параметром.
Также можно сделать protected конструктор в базовом классе и вызывать его также, с помощью цепочки вызовов
`public Resistor(double value): base(value) {}`

iKalentii commented on Lab1Model/IComponent.cs:
```diff
@@ -0,0 +1,6 @@
+{
+    public double Value { get; set; }
+
+    public double GetImpedance(double freq);
```
Возвращаемое значение должно быть комплексным

iKalentii commented on Lab1Model/AbstractComponent.cs:
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+
+public abstract class Component : IComponent
```
По стандарту RSDN абстрактные классы должны именоваться с постфиксом Base - чтобы сразу было понятно, что нельзя создать его экземпляр. Надо только наследовать.

iKalentii commented on ConsoleLoader/Program.cs:
```diff
@@ -0,0 +1,80 @@
+                break;
+
+            default:
+                Console.Write("Bad component type");
```
Обработка входных данных выполнена некорректно. Если данные введены некорректно, пользователю должна предоставляться возможность ввода заново с описанием - что пользователь конкретно ввёл некорректно и что ему необходимо ввести, чтобы подобная ошибка не повторилась.

iKalentii commented on ConsoleLoader/Program.cs:
```diff
@@ -0,0 +1,80 @@
+
+public static class Program
+{
+    public static void Main()
```
Длинный метод Main, в идеале (ооочень редко такое бывает =) ) размер метода не должен превышать размер экрана. Здесь будет правильней побить Main на более мелкие методы и повыносить части кода туда.

iKalentii commented on ConsoleLoader/Program.cs:
```diff
@@ -0,0 +1,80 @@
+
+        try
+        {
+            freq = Convert.ToDouble(Console.ReadLine().Replace('.', ','));
```
Тут из строки может быть получен NaN, если его ввести. Правильней проверить значение после конвертации и, если оно не вписывается в область допустимых значений, просить пользователя ввести его ещё раз, рассказав - что пошло не так.

iKalentii replied:
Тут же может быть получено отрицательное число. Корректнее его обработать уже на этапе ввода, считая, что такие данные не должны дойти до твоего объекта. Если уже дойдут - тогда сработает второй предохранитель в виде исключения в конструкторе.

iKalentii commented on ConsoleLoader/Program.cs:
```diff
@@ -0,0 +1,80 @@
+{
+    public static void Main()
+    {
+        Console.Write("Enter component type (R, L, C): ");
```
Хорошим тоном для пользовательского интерфейса тут будет зациклить запрос и сделать отдельную команду для выхода и закрытия программы

iKalentii commented on ConsoleLoader/ConsoleLoader.cs:
```diff
@@ -0,0 +1,158 @@
+    /// Делегат для передачи информационных сообщений
+    /// </summary>
+    /// <param name="message">Информационное сообщение</param>
+    public delegate void Printer(string message);
```
Вместо отдельно делегата можно использовать анонимный делегат. Удобнее и можно сразу писать в качестве входного параметра метода.
`Action<inputType1, inputType2,...> ` - где inputTypeN входной тип функции

iKalentii commented on ConsoleLoader/ConsoleLoader.cs:
```diff
@@ -0,0 +1,158 @@
+                    "Введите число вида X.Y или X,Y, где X и Y - наборы цифр.");
+            }
+
+            if ((value < double.MinValue) || (value > double.MaxValue))
```
Есть у меня ощущение, что тебе просто не получится ввести такие числа, т.к. они выходят за тип. Скорее всего там будут или +- бесконечности или NaN. Поэтому лучше и проверять на них сразу. double.NaN, double.PositiveInfinite и т.п.

iKalentii commented on ConsoleLoader/ConsoleLoader.cs:
```diff
@@ -0,0 +1,158 @@
+        public static void AskComponentValue(
+            in ComponentBase cmp, Printer printer)
+        {
+            if (cmp is Resistor)
```
Тут можно вместо кучи if-ов использовать pattern-matching в switch-case получается лаконичнее.

iKalentii commented on ConsoleLoader/Main.cs:
```diff
@@ -0,0 +1,84 @@
+		Console.WriteLine("Программа для вычисления\n" +
+			"комплексного сопротивления радиокомпонентов");
+
+		// Основной цикл программы
```
Вот начиная отсюда и ниже - ты перестарался с комментариями =)
Комментировать необходимо не что ты делаешь, а почему. И комментировать стоит только тогда, когда написанные код использует какие-то слабо задокументированные ходы или, когда ты применяешь не стандартные программистские решения. В любых других случаях комментарии только затрудняют понимание кода.
У меня правило примерно такое, если тебе приходится больше чем на 30 секунд зависать на участке кода - его надо или отрефакторить или написать внятный комментарий.

iKalentii replied:
Собственно - большая часть комментариев тут тривиальны и понятны из кода.

iKalentii commented on ConsoleLoader/Main.cs:
```diff
@@ -0,0 +1,84 @@
+
+			// На основании введенной строки,
+			// создать объект радиокомпонента
+			ComponentBase cmp = ConsoleLoader.GetComponent(
```
Сильно сократил название не по стандарту.

iKalentii commented on ConsoleLoader/Main.cs:
```diff
@@ -0,0 +1,84 @@
+			string inputStr = Console.ReadLine();
+
+			// Если введена "Q" или "q" - выйти из программы
+			if (inputStr.ToUpper() == "Q") { return; }
```
Тут можно было ввести константу, назвать её как-нибудь типа upperCaseExitCharacter и тогда можно было бы не писать коммент.

iKalentii commented on ConsoleLoader/Main.cs:
```diff
@@ -0,0 +1,84 @@
+			// радиокомпонента и считывает введенную строку
+
+			ConsoleLoader.AskComponentValue(in cmp, Console.Write);
+			inputStr = Console.ReadLine();
```
Старайся не использовать одну переменную в нескольких контекстах, т.к. можно забыть переинициализировать её и получишь баг. Лучше заводить разные переменные и локализовывать области видимости, чтобы эта переменная не болталась больше чем надо.

TemZZZ replied:
Не до конца понял, что имеется ввиду. Вместо одной переменной inputStr делать несколько конкретных: спросили у пользователя тип компонента - в переменную string componentTypeString, спросили значение - string valueString, спросили частоту - string frequencyString, правильно понимаю?

iKalentii replied:
Да, всё правильно.

iKalentii commented on Lab1Model/ComponentBase.cs:
```diff
@@ -0,0 +1,85 @@
+
+            set
+            {
+                // Если значение частоты больше или равно нулю,
```
Лишний коммент

iKalentii commented on Lab1Model/ComponentBase.cs:
```diff
@@ -0,0 +1,85 @@
+        /// значения частоты</exception>
+        public Complex GetImpedance(double freq)
+        {
+            // Если значение частоты меньше нуля,
```
Лишний коммент.

iKalentii commented on Lab1Model/Resistor.cs:
```diff
@@ -0,0 +1,48 @@
+	namespace PassiveComponents
+	{
+		/// <summary>
+		/// ����� ���������
```
Чёт кодировка поехала.

iKalentii commented on ConsoleLoader/ConsoleLoader.cs:
```diff
@@ -0,0 +1,158 @@
+        /// <see cref="ComponentBase"/></param>
+        /// <param name="printer">Делегат для передачи запросов</param>
+        public static void AskComponentValue(
+            in ComponentBase cmp, Printer printer)
```
Сильно сократил название переменно, не по стандарту.

iKalentii commented on ConsoleLoader/ConsoleLoader.cs:
```diff
@@ -0,0 +1,158 @@
+        /// <returns>Объект класса <see cref="Resistor"/>,
+        /// <see cref="Inductor"/> или
+        /// <see cref="Capacitor"/></returns>
+        public static ComponentBase GetComponent(
```
Вот это хороший подход. Классы, в которых собирают знания об иерархии наследования и которые генерят эти объекты называют Фабриками. И это общеиспользуемый паттерн "Фабрика".

iKalentii commented on ConsoleLoader/ConsoleLoader.cs:
```diff
@@ -0,0 +1,169 @@
+    public static class ConsoleLoader
+    {
+        /// <summary>Символ резистора</summary>
+        const string resistorCharacter = "R";
```
Если эти вещи используются больше чем в одном месте, тогда есть смысл их выносить на уровень класса, если нет, то лучше оставить в методе.

iKalentii commented on ConsoleLoader/Main.cs:
```diff
@@ -0,0 +1,64 @@
+﻿using System;
+
```
Блок using можно писать без пустых строк, вместе.

iKalentii commented on ConsoleLoader/Main.cs:
```diff
@@ -0,0 +1,64 @@
+		{
+			ComponentBase component;
+
+			{
```
Зачем выделил отдельный блок кода? Лучше его убрать, а component проинициализировать сразу.

iKalentii commented on ConsoleLoader/Main.cs:
```diff
@@ -0,0 +1,64 @@
+			{
+				Console.Write("\nДля выхода из программы введите Q (q)\n" +
+				"или введите тип радиокомпонента R (r), L (l) или C (c): ");
+
```
Много пустых строк, визуально удленнил код. На мой взгляд у тебя  тут используется примерно такая схема - сообщение, считывание с инициализацией, проверка условия и следующий блок такой же. Собственно так и можно было разделять строки. Такая группировка показала эти логические блоки и ход работы программы стал бы восприниматься проще. А так тебе приходится сначала прочитать строки, они не сгрупированы, поэтому тебе приходится всю картину простраивать в голове, что несколько сложнее. Если можно делать такие блоки оформлением кода - это лучше с точки зрения восприятия.

Commit:
feature(MainForm): тип списка radioComponents изменен с List на BindingList

feature(MainForm): изменен модификатор доступа к списку radioComponents с private на internal static

Commit:
style(MainForm): добавлен модификатор доступа к объекту addRadioComponentForm

Commit:
style(MainForm): добавлено ключевое слово readonly к объекту addRadioComponentForm

Commit:
style(AddRadioComponentForm.RadioButton_CheckedChanged): проверка sender на null заменена сопоставлением шаблонов

Commit:
feature(AddRadioComponentForm): добавлен метод проверки строки на соответствие положительному вещественному ToPositiveDouble

Commit:
feature(AddRadioComponentForm): добавлен метод Messager для отображения предупреждений пользователю

Commit:
feature(AddRadioComponentForm): реализовано добавление радиокомпонента в AddRadioComponentButton_Click

Commit:
feature(MainForm): добавлена инициализация списка radioComponents в конструкторе

Commit:
style(MainForm): добавлен комментарий к полю addRadioComponentForm

Commit:
style(AddRadioComponentForm.GenerateRandomDataButton_Click): удалены лишние пустые строки

Commit:
fix(Capacitor): исправлены константы capacitorUnit, capacitorType, capacitorQuantity

Commit:
feature(MainForm): добавлено связывание списка radioComponents с radioComponentsDataGridView в конструкторе класса

Commit:
style(MainForm): у пользователя появилась возможность менять местами столбцы в radioComponentsDataGridView

Commit:
feature(MainForm): добавлено удаление строк кнопкой deleteRadioComponentButton

Commit:
feature(MainForm): пользователь теперь может выделить только всю строку целиком в radioComponentsDataGridView

Commit:
feature(MainForm): добавлен метод FormatRadioComponentsDataGridView

Commit:
feature(MainForm): добавлен вызов метода FormatRadioComponentsDataGridView в конструктор

Commit:
style(MainForm.FormatRadioComponentsDataGridView): добавлен комментарий

Commit:
style(MainForm): изменена геометрия формы

Commit:
feature(Lab1View): добавлен класс SortableBindingList

feature(MainForm): тип списка radioComponents изменен с BindingList на SortableBindingList

Commit:
feature(solution Lab1): добавлен проект PositiveDoubleTextBoxLib

- реализован класс PositiveDoubleTextBox, наследующий RegexTextBox
- класс содержит константу шаблона регулярного выражения положительных вещественных чисел positiveDoublePattern (взят из класса AddRadioComponentForm)
- класс содержит метод ToPositiveDouble (взят из класса AddRadioComponentForm)
- класс содержит метод Messager (взят из класса AddRadioComponentForm)

Commit:
feature(PositiveDoubleTextBox.ToPositiveDouble): messager теперь по умолчанию null

- теперь если не указать messager, сообщения об ошибках не будут передаваться

Commit:
feature(AddRadioComponentForm): valueRegexTextBox заменен на valuePositiveDoubleTextBox

- удалена константа шаблона positiveDoublePattern
- удалено присвоение шаблона элементу valueRegexTextBox (valueRegexTextBox теперь нет на форме)
- удален метод ToPositiveDouble
- удален метод Messager
- в методе GenerateRandomDataButton_Click valueRegexTextBox изменен на valuePositiveDoubleTextBox
- в методе AddRadioComponentButton_Click переписан вызов метода ToPositiveDouble на вызов PositiveDoubleTextBox.ToPositiveDouble без messager
- в методе AddRadioComponentButton_Click valueRegexTextBox изменен на valuePositiveDoubleTextBox

style(AddRadioComponentForm): пересортированы using

Commit:
feature(MainForm): изменена форма

- добавлены кнопки для сохранения в файл/загрузки из файлов радиокомпонентов
- добавлено поле для ввода частоты
- добавлено поле для вывода импеданса
- добавлена кнопка для поиска

Commit:
feature(PositiveDoubleTextBox): добавлен метод GetValue

feature(MainForm): добавлен расчет и вывод импеданса RadioComponentsDataGridView_RowEnter

Commit:
feature(MainForm): добавлен метод ComplexToText

Commit:
feature(MainForm.ComplexToText): добавлен формат строки

Commit:
feature(MainForm.RadioComponentsDataGridView_RowEnter):
добавлена проверка на множественное выделение строк

Commit:
feature(MainForm): добавлены обработчики

- FrequencyPositiveDoubleTextBox_LostFocus
- RadioComponentsDataGridView_RowsAdded

Commit:
style(MainForm): добавлены комментарии к методам

Commit:
feature(Lab1View): изменены стартовые позиции форм

- MainForm запускается по центру экрана
- AddRadioComponentForm запускается по центру MainForm

Commit:
fix(AddRadioComponentForm.AddRadioComponentButton_Click): исправлена ошибка

- пользователю не выводилось сообщение о недопустимости пустого поля значения радиокомпонента, когда изначально у этого поля не было фокуса
- добавлен PositiveDoubleTextBox.Messager в параметры метода PositiveDoubleTextBox.ToPositiveDouble

Commit:
fix(MainForm.RadioComponentsDataGridView_RowEnter): исправлена ошибка

- после удаления последней строки таблицы radioComponentsDataGridView переменная index принимала значение больше, чем элементов в коллекции radioComponents

Commit:
fix(MainForm.DeleteRadioComponentButton_Click): исправлены ошибки, возникающие при удалении строк

- исправлена ошибка при удалении промежуточных строк
- исправлена ошибка при удалении последний строки
- добавлены комментарии в методе, поясняющие ошибки и их исправление

Commit:
style(MainForm): исправлен комментарий в конструкторе

Commit:
style(PositiveDoubleTextBox.cs): удалены неиспользуемые using

Commit:
style(MainForm): изменены Anchor'ы элементов управления

Commit:
feature(MainForm): добавлены openFileDialog и saveFileDialog на форму

feature(MainForm): добавлены пустые обработчики SaveToFileButton_Click и LoadFromFileButton_Click

Commit:
refactor(MainForm.SetupRadioComponentsDataGridView): переименован из MainForm.FormatRadioComponentsDataGridView

Commit:
feature(MainForm): добавлена структура NameHeaderTextPair

- структура используется для редактирования внешнего вида таблицы DataGridView

feature(MainForm.SetupRadioComponentsDataGridView): метод переписан с применением структур NameHeaderTextPair

style(MainForm.SetupRadioComponentsDataGridView): обновлен комментарий

Commit:
style(MainForm.cs): блоки if оформлены по MSDN

Commit:
style(RadioComponentBase.cs): код оформлен в соответствие с MSDN

Commit:
style(AddRadioComponentForm.cs): код оформлен в соответствие с MSDN

Commit:
style(SortableBindingList.cs): код оформлен в соответствие с MSDN

Commit:
style(PositiveDoubleTextBox.cs): код оформлен в соответствие с MSDN

Commit:
style(RegexTextBox.cs): добавлены XML-комментарии, код оформлен по MSDN

Commit:
style(PositiveDoubleTextBox.cs): добавлены XML-комментарии к конструктору, к OnLostFocus

Commit:
style(RegexTextBox.cs): исправлен комментарий к конструктору

Commit:
style(MainForm.cs): добавлены XML-комментарии к классу и конструктору

Commit:
style(AddRadioComponentForm.cs): добавлены XML-комментарии

Commit:
feature(RadioComponentBase): добавлены атрибуты для возможности сериализации объектов Resistor, Inductor и Capacitor

Commit:
style(RadioComponentBase): удален комментарий про атрибуты сериализации

- этот комментарий не позволял корректно отбражать описание класса в VisualStudio

Commit:
feature(Lab1Model): добавлен класс RadioComponentIO

Commit:
style(RadioComponentBase): добавлены фигурные скобки к блокам if

Commit:
style(AddRadioComponentForm): добавлены фигурные скобки в блоках if

Commit:
style(MainForm): добавлены фигурные скобки в блоках if

Commit:
style(SortableBindingList): добавлены фигурные скобки в блоках if и foreach

Commit:
style(RadioComponentBase): удалены очевидные комментарии

- в методе set свойства Value
- в методе GetImpedance

Commit:
feature(MainForm): удалены комментарии

- очевидный XML-комментарий из конструктора
- в методе DeleteRadioComponentButton_Click XML-комментарии заменены обычными

Commit:
style(PositiveDoubleTextBox): XML-комментарий в конструкторе переделан в простой

Commit:
style(AddRadioComponentForm.AddRadioComponentButton_Click): исправлен комментарий

Commit:
feature(MainForm): установлен фиксированный минимальный размер формы

Commit:
feature(AddRadioComponentForm): генератор случайных чисел randomIntGenerator перенесен в метод GenerateRandomDataButton_Click

Commit:
feature(MainForm): форма addRadioComponentForm теперь создается только при нажатии кнопки addRadioComponentButton

Commit:
feature(SortableBindingList): изменен уровень доступности с internal на public

Commit:
feature(MainForm.radioComponents): изменен уровень доступности поля с internal на public

Commit:
refactor(MainForm): RadioComponents переименован из radioComponents

Commit:
feature(MainForm.RadioComponents): удалено ключевое слово static

Commit:
feature(MainForm): коллекция RadioComponents объявлена как internal и инициализируется сразу

- инициализация RadioComponents  из конструктора класса удалена

Commit:
feature(RadioComponentBase.cs): добавлено событие создания нового радиокомпонента

- добавлен делегат RadioComponentCreatedEventHandler
- добавлен и реализован класс данных события RadioComponentCreatedEventArgs
- в класс AddRadioComponentForm добавлено поле события RadioComponentCreated
- в метод AddRadioComponentButton_Click добавлен вызов события RadioComponentCreated с передачей созданного радиокомпонента

Commit:
feature(MainForm): реализован обработчик события создания нового радиокомпонента OnRadioComponentCreated

feature(MainForm.AddRadioComponentButton_Click): для события addRadioComponentForm.RadioComponentCreated зарегистрирован обработчик OnRadioComponentCreated

Commit:
style(Capacitor): исправлены комментарии у конструкторов

style(Capacitor): добавлены inheritdoc-комментарии к свойствам и к методу CalcImpedance

Commit:
style(Inductor): исправлены комментарии у конструкторов

style(Inductor): добавлены inheritdoc комментарии свойствам и методу CalcImpedance

Commit:
style(RadioComponentBase): исправлен комментарий у конструктора

style(RadioComponentBase): добавлены inheritdoc комментарии свойствам

Commit:
style(Resistor): исправлены комментарии у конструкторов

style(Resistor): добавлены inheritdoc комментарии свойствам и методу CalcImpedance

Commit:
style(AddRadioComponentForm): исправлены/добавлены комментарии

- добавлен комментарий к CancelButton_Click
- исправлен комментарий к AddRadioComponentButton_Click

Commit:
style(MainForm): исправлен комментарий к AddRadioComponentButton_Click

Commit:
style(SortableBindingList): добавлены inheritdoc комментарии

Commit:
fix(PositiveDoubleTextBox): исправлена ошибка некорректного преобразования строк в число из-за регионального стандарта

- реализован метод DotAndCommaToNumberDecimalSeparator, заменяющий точки и запятые на региональный десятичный разделитель
- метод ToPositiveDouble исправлен с учетом корректировки строки методом DotAndCommaToNumberDecimalSeparator

Commit:
style(Capacitor.CalcImpedance): действительная часть комплексного числа переписана через константу zeroRealPart

Commit:
style(Inductor.CalcImpedance): действительная часть комплексного числа переписана через константу zeroRealPart

Commit:
style(ConsoleLoader): удалены комментарии из методов, в блоках if добавлены фигурные скобки

Commit:
style(ConsoleLoader): удалена пустая строка между using

Commit:
style(Main): удален очевидный комментарий

Commit:
style(Main): код оформлен по RSDN

Commit:
feature(ConsoleLoader): добавлен метод DotAndCommaToNumberDecimalSeparator

Commit:
feature(StringToDouble): метод StringToDouble переписан с учетом DotAndCommaToNumberDecimalSeparator

Commit:
fix(ConsoleLoader): исправлен метод PrintComplex

- неверно выводилась бесконечность методом PrintComplex

Commit:
refactor(ConsoleLoader.GetRadioComponent): переименован из ConsoleLoader.GetComponent

Commit:
refactor(ConsoleLoader.AskRadioComponentValue): переименован из ConsoleLoader.AskComponentValue

Commit:
style(Main): длина строк кода теперь не более 78 символов

Commit:
feature(Main): переписана логика программы

- запрос типа радиокомпонента осуществляется в GetRadioComponentLoop
- запрос значения физической величины осуществляется в GetRadioComponentValueLoop
- запрос частоты осуществляется GetFrequencyLoop

Commit:
feature(RadioComponentIO): переписан класс

- удалены методы WriteXml
- удален метод GetFile
- добавлен метод SerializeAndWriteXml
- добавлен метод GetFileStream
- добавлен метод GetStreamReader
- добавлен метод ReadXmlAndDeserialize

Commit:
refactor(XmlReaderWriter): переименован из RadioComponentIO

Commit:
feature(XmlReaderWriter.cs): файл перенесен из проекта Lab1Model в проект Lab1View

Commit:
feature(XmlReaderWriter.cs): изменено пространство имен с Lab1Model на Lab1View

Commit:
style(AddRadioComponentForm.cs): код класса AddRadioComponentForm поставлен первым в файле для возможности открывания формы в конструкторе

Commit:
feature(Lab1View): добавлена форма выбора параметров сохранения радиокомпонентов SetRadioComponentSaveOptionForm

- добавлено перечисление опций сохранения RadioComponentSaveOption
- добавлено событие подтверждения выбора опции RadioComponentReadyToSaveEventHandler
- реализован класс данных события RadioComponentReadyToSaveEventArgs
- реализовано подтверждения выбора опции сохранения кликом по кнопке OkButton_Click
- реализовано закрытие формы кликом по кнопке CancelButton_Click

Commit:
feature(MainForm): добавлена возможность сохранения радиокомпонентов

- метод SaveToFileButton_Click открывает окно для выбора опций сохранения
- метод SaveToFile сохраняет все или выделенную часть компонентов в файл

Commit:
feature(XmlReaderWriter): класс переделан в нестатический

feature(MainForm.SaveToFile): для вызова методов XmlReaderWriter создан объект xmlWriter

Commit:
feature(SetRadioComponentSaveOptionForm.OkButton_Click): после события RadioComponentReadyToSave форма закрывается

Commit:
style(MainForm): добавлены комментарии

- добавлены комментарии к методу GetRadioComponentsToSave
- добавлены комментарии к методу SaveToFile
- добавлены комментарии к методу SaveToFileButton_Click

Commit:
style(SetRadioComponentSaveOptionForm): изменен комментарий к событию RadioComponentReadyToSave

Commit:
feature(XmlReaderWriter): добавлено событие FileSaved

- добавлено возникновение события FileSaved в методе SerializeAndWriteXml

Commit:
feature(XmlReaderWriter): удалено событие FileSaved

Commit:
feature(Lab1View): добавлена форма выбора параметров загрузки радиокомпонентов SetRadioComponentLoadOptionForm

- добавлено перечисление опций загрузки RadioComponentLoadOption
- добавлено событие подтверждения выбора опции RadioComponentReadyToLoadEventHandler
- реализован класс данных события RadioComponentReadyToLoadEventArgs
- реализовано подтверждение выбора опции загрузки кликом по кнопке OkButton_Click
- реализовано закрытие формы кликом по кнопке CancelButton_Click

Commit:
feature(MainForm): добавлена возможность загрузки радиокомпонентов

- метод LoadFromFileButton_Click открывает окно для выбора опций загрузки
- метод LoadFromFile добавляет загруженные радиокомпоненты в конец таблицы или заменяет всю таблицу новыми элементами

Commit:
feature(MainForm): добавлен метод SetupFileDialogs для редактирования внешнего вида окон загрузки и сохранения файлов

- в конструкторе класса метод SetupFileDialogs вызывается

Commit:
style(MainForm): удалены неиспользованные using

Commit:
style(SetRadioComponentLoadOptionForm): зафиксирован размер формы

Commit:
style(SetRadioComponentLoadOptionForm): деактивирована кнопка разворачивания формы на весь экран

Commit:
feature(AddRadioComponentForm): значение resistorRadioButton.Checked = true по умолчанию задано в конструкторе форм

- из конструктора класса установка значения по умолчанию resistorRadioButton.Checked = true удалена

Commit:
style(PositiveDoubleTextBox.Messager): исправлен комментарий

Commit:
feature(MainForm): PositiveDoubleTextBox.Messager определен как поле-делегат ErrorMessager

- в методе SaveToFile вызов PositiveDoubleTextBox.Messager заменен на ErrorMessager
- в методе SaveToFileButton_Click вызов PositiveDoubleTextBox.Messager заменен на ErrorMessager
- в методе LoadFromFile вызов PositiveDoubleTextBox.Messager заменен на ErrorMessager

Commit:
style(MainForm): Text изменен с "MainForm" на "Коллекция радиокомпонентов"

Commit:
feature(AddRadioComponentForm): возвращена установка значения по умолчанию resistorRadioButton.Checked = true в конструктор класса

- в конструкторе форм значение resistorRadioButton.Checked установлено в false

Commit:
fix(PositiveDoubleTextBox): исправлена ошибка - переопределенный метод OnLostFocus блокировал добавление новых обработчиков события LostFocus

- удалено ключевое слово override у OnLostFocus
- изменен уровень доступности с protected на private у OnLostFocus
- в конструкторе класса обработчик OnLostFocus добавляется к событию LostFocus как this.LostFocus += OnLostFocus

Commit:
feature(MainForm): изменена функциональность

- в методе DeleteRadioComponentButton_Click остался только цикл удаления выделенных строк
- метод RadioComponentsDataGridView_RowEnter удален
- функциональность метода RadioComponentsDataGridView_RowEnter перенесена в добавленный метод RadioComponentsDataGridView_SelectionChanged

Commit:
feature(MainForm.FrequencyPositiveDoubleTextBox_LostFocus): изменена функциональность

- теперь вместо обновления radioComponentsDataGridView вызывается RadioComponentsDataGridView_SelectionChanged пересчитывающий импеданс выделенного радиокомпонента
- в конструкторе MainForm событию frequencyPositiveDoubleTextBox.LostFocus добавлен обработчик FrequencyPositiveDoubleTextBox_LostFocus;

Commit:
feature(Lab1View): добавлена форма поиска радиокомпонентов SearchRadioComponentForm

Commit:
feature(MainForm): реализовано создание и открывание формы поиска радиокомпонентов кликом по кнопке SearchButton_Click

Commit:
style(PositiveDoubleTextBox): удалены указатели this

feature(PositiveDoubleTextBox): добавлена константа _defaultText

- элемент по умолчанию при создании заполняется текстом _defaultText
- если после редактирования элемент пустой, то он заполняется текстом _defaultText в методе OnLostFocus

Commit:
feature(SearchRadioComponentForm): реализовано заполнение комбобокса типами радиокомпонентов

feature(SearchRadioComponentForm): реализовано закрывание формы кликом по кнопке CancelButton_Click

Commit:
feature(MainForm): реализовано открывание формы поиска кнопкой SearchButton_Click

Commit:
feature(Lab1View): добавлена форма поиска радиокомпонентов SearchRadioComponentForm

- реализовано закрывание формы кнопкой CancelButton_Click
- реализовано заполнение комбобокса типами радиокомпонентов
- реализовано информирование пользователя об изменении фильтра поиска

Commit:
feature(MinForm): список радиокомпонентов инкапсулирован

- создано приватное поле _radioComponents
- доступ к списку осуществляется через свойство RadioComponents

Commit:
style(AddRadioComponentForm): удален тип  делегата RadioComponentCreatedEventHandler

- событие public event RadioComponentCreatedEventHandler RadioComponentCreated переписано как public event EventHandler<RadioComponentCreatedEventArgs> RadioComponentCreated

Commit:
style(AddRadioComponentForm): класс RadioComponentCreatedEventArgs перенесен в файл RadioComponentCreatedEventArgs.cs

Commit:
feature(SearchRadioComponentForm): свойство RadioComponents переделано в автосвойство с единственным методом get

feature(SearchRadioComponentForm): конструктор переделан в конструктор с параметром

- теперь в конструктор обязательно нужно передать список радиокомпонентов radioComponents
- в конструкторе свойству RadioComponents присваивается параметр radioComponents

feature(SearchRadioComponentForm): реализована реакция на изменения в списке радиокомпонентов в методе OnRadioComponentsChanged

- событию RadioComponents.ListChanged добавлен обработчик OnRadioComponentsChanged в конструкторе класса

Commit:
feature(MainForm): изменено свойство RadioComponents

- свойство RadioComponents переделано в автосвойство
- модификатор доступа изменен на private

feature(MainForm.SearchButton_Click): в конструктор формы searchRadioComponentForm теперь передается параметр RadioComponents

Commit:
feature(SearchRadioComponentForm): добавлен метод GetFilteredRadioComponentsIndices

- GetFilteredRadioComponentsIndices возвращающает индексы радиокомпонентов, удовлетворяющих условиям поиска

Commit:
feature(SearchRadioComponentForm.OnRadioComponentsChanged): добавлена проверка на пустой список

- если список пустой, то кнопка поиска деактивируется и пользователю сообщается о невозможности поиска

Commit:
feature(Lab1View): добавлен класс данных события завершения поиска

- добавлен файл SearchFinishedEventArgs.cs
- в файле реализован класс данных события завершения поиска SearchFinishedEventArgs

Commit:
feature(SearchRadioComponentForm): добавлено событие, происходящее после завершения поиска SearchFinished

Commit:
feature(SearchRadioComponentForm): реализован поиск кликом по кнопке SearchRadioComponentsButton_Click

Commit:
feature(MainForm): реализовано выделение найденых радиокомпонентов OnSearchFinished

Commit:
fix(RegexTextBox.OnTextChanged): переопределенный OnTextChanged не позволял обрабатывать событи TextChanged другим подключенным обработчикам

- изменена сигнатура метода с protected override ... на private ...
- удален вызов base.OnTextChanged(e), если _regex is null
- в конструкторе RegexTextBox к событию TextChanged добавляется обработчик OnTextChanged

Commit:
style(RegexTextBox.OnTextChanged): изменет комментарий, в теле метода одиночный return написан без фигурных скобок {}

Commit:
style(ConsoleLoader.DotAndCommaToNumberDecimalSeparator): sep переименован в regionalNumberDecimalSeparator

Commit:
style(ConsoleLoader.DotAndCommaToNumberDecimalSeparator): изменен комментарий

Commit:
style(ConsoleLoader.StringToDouble): в комментарии стандартные типы данных записаны без тэгов

Commit:
style(ConsoleLoader.IsPositive): в комментарии стандартные типы данных записаны без тэгов

Commit:
style(ConsileLoader.cs): дтрективы using пространства имен System записаны в алфавитном порядке

Commit:
style(MainApp.GetRadioComponentLoop): null в комментарии записан без тэгов

Commit:
style(MainApp.GetRadioComponentValueLoop): стандартные типы данных в комментарии записаны без тэгов

Commit:
style(MainApp.GetFrequencyLoop): стандартные типы данных в комментарии записаны без тэгов

Commit:
style(Capacitor): свойства Unit, Type и Quantity сразу возвращают литеральные константы

- константы capacitorUnit, capacitorType и capacitorQuantity удалены

Commit:
style(Inductor): свойства Unit, Type и Quantity сразу возвращают литеральные константы

- константы inductorUnit, inductorType и inductorQuantity удалены

Commit:
style(Resistor): свойства Unit, Type и Quantity сразу возвращают литеральные константы

- константы resistorUnit, resistorType и resistorQuantity удалены

Commit:
style(IRadioComponent.Value): строки комментария теперь не занимают более 78 символов

Commit:
style(PositiveDoubleTextBox.cs): директивы using пространства имен System отсортированы по алфавиту

Commit:
style(PositiveDoubleTextBox.DotAndCommaToNumberDecimalSeparator): изменен комментарий

style(PositiveDoubleTextBox.DotAndCommaToNumberDecimalSeparator): sep переименован в regionalNumberDecimalSeparator

Commit:
style(PositiveDoubleTextBox.OnLostFocus): изменен комментарий

Commit:
feature(AddRadioComponentForm.AddRadioComponentButton_Click): добавлена проверка подписанного обработчика на null перед вызовом события RadioComponentCreated

- событие вызывается RadioComponentCreated?.Invoke(...)

Commit:
style(SearchFinishedEventArgs): изменен комментарий к конструктоу

Commit:
feature(SetRadioComponentLoadOptionForm.OkButton_Click): добавлена проверка подписанного обработчика на null перед вызовом события RadioComponentReadyToLoad

- событие вызывается RadioComponentReadyToLoad?.Invoke(...)

Commit:
feature(SetRadioComponentLoadOptionForm): изменено объявление события RadioComponentReadyToLoad

- событие объявлено как делегат EventHandler<RadioComponentReadyToLoadEventArgs>

Commit:
style(SetRadioComponentLoadOptionForm.cs): исправлен комментарий к классу RadioComponentReadyToLoadEventArgs

Commit:
feature(SetRadioComponentLoadOptionForm.cs): удален делегат RadioComponentReadyToLoadEventHandler

Commit:
feature(SetRadioComponentLoadOptionForm.cs): класс RadioComponentReadyToLoadEventArgs перемещен в файл RadioComponentReadyToLoadEventArgs.cs

Commit:
feature(SetRadioComponentLoadOptionForm.cs): перечисление RadioComponentLoadOption перемещено в файл RadioComponentReadyToLoadEventArgs.cs

Commit:
feature(SetRadioComponentSaveOptionForm): изменено объявление события RadioComponentReadyToSave

- событие объявлено как EventHandler<RadioComponentReadyToSaveEventArgs>

Commit:
feature(SetRadioComponentSaveOptionForm.cs): удален делегат RadioComponentReadyToSaveEventHandler

Commit:
style(SetRadioComponentSaveOptionForm.cs): изменен комментарий к классу RadioComponentReadyToSaveEventArgs

Commit:
style(SetRadioComponentSaveOptionForm.cs): класс RadioComponentReadyToSaveEventArgs перемещен в файл RadioComponentReadyToSaveEventArgs.cs

Commit:
style(SetRadioComponentSaveOptionForm.cs): перечисление RadioComponentSaveOption перемещено в файл RadioComponentReadyToSaveEventArgs.cs

Commit:
feature(SetRadioComponentSaveOptionForm.OkButton_Click): добавлена проверка подписанного обработчика на null перед вызовом события RadioComponentReadyToSave

- событие вызывается RadioComponentReadyToSave?.Invoke(...)

Commit:
style(XmlReaderWriter.cs): директивы using отсортированы по алфавиту

Commit:
feature(MainForm): в конструкторе класса на событие frequencyPositiveDoubleTextBox.LostFocus подписан RadioComponentsDataGridView_SelectionChanged

Commit:
feature(MainForm): удалена подписка обработчика FrequencyPositiveDoubleTextBox_LostFocus на frequencyPositiveDoubleTextBox.LostFocus

Commit:
feature(MainForm): удален метод FrequencyPositiveDoubleTextBox_LostFocus

Commit:
style(MainForm.ComplexToText): исправлен комментарий

Commit:
fix(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): поиск выполнялся неправильно

- если фильтрации меньше, больше, равно давали пустые множества, то фильтр по типу радиокомпонента не сбрасывался
- добавлена проверка включенных фильтров после фильтрации по типу: если ни один фильтр из больше, меньше, равно не включен, то сразу возвращаются индексы отфильтрованных компонентов по типу
- удалена проверка на пустое множество перед последним return

Commit:
style(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): переформатированы переносы длинных инструкций

Commit:
style(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): добавлены комментарии в метод

Commit:
feature(XmlReaderWriter.SerializeAndWriteXml): добавлен отлов исключения SerializeAndWriteXml

- при возникновении исключения SerializeAndWriteXml через messager отправляется сообщение о невозможности сериализации объекта
- при возникновении другого типа исключения через messager отправляется сообщение исключения, и исключение создается повторно

Commit:
feature(XmlReaderWriter.ReadXmlAndDeserialize): добавлен отлов исключения SerializeAndWriteXml

- при возникновении исключения SerializeAndWriteXml через messager отправляется сообщение о невозможности десериализации в объект
- при возникновении другого типа исключения через messager отправляется сообщение исключения, и исключение создается повторно

Commit:
style(XmlReaderWriter.GetFileStream): null в комментарии записан без тэгов

Commit:
style(XmlReaderWriter.GetStreamReader): null в комментарии записан без тэгов

Commit:
style(XmlReaderWriter.ReadXmlAndDeserialize): null в комментарии записан без тэгов

Commit:
feature(Lab1View): добавлен класс вызова функции и обработки возможных исключений ExceptionHandler

- реализован обобщенный метод вызова функции и обработки возможных исключений CallFunction

Commit:
fix(ExceptionHandler.CallFunction): в блоке catch всегда обрабатывается первое исключение из перечислителя exceptionToMessageMap

- в цикле foreach блока catch в операторе сравнения типов исключений if (!(exceptionToMessage.Key.GetType() != e.GetType())) удален оператор !

Commit:
feature(XmlReaderWriter): методы GetFileStream и GetStreamReader переписаны через вызов ExceptionHandler.CallFunction(...)

Commit:
feature(ExceptionHandler.CallFunction): изменен тип входного параметра

- тип параметра exceptionToMessageMap изменен с IEnumerable<KeyValuePair<Exception, string>> на IEnumerable<KeyValuePair<Type, string>>
- имя параметра exceptionToMessageMap изменено на exceptionTypeToMessageMap
- в операторе сравнения в цикле foreach вызов exceptionToMessage.Key.GetType() изменен на exceptionToMessage.Key
- исправлен комментарий

Commit:
refactor(XmlReaderWriter): словарь StreamExceptionToMessageMap переименован в StreamExceptionTypeToMessageMap

feature(XmlReaderWriter): изменен словарь StreamExceptionTypeToMessageMap

- теперь словарь заполняется не парами {Exception obj, string message}, а парами {typeof(<класс исключения>), string message}

Commit:
style(XmlReaderWriter.StreamExceptionTypeToMessageMap): добавлен комментарий

Commit:
feature(XmlReaderWriter.StreamExceptionTypeToMessageMap): изменено сообщение для типа исключения UnauthorizedAccessException

Commit:
feature(XmlReaderWriter.StreamExceptionTypeToMessageMap): добавлен элемент

- добавлен {typeof(IOException), string messageOnIOException}

Commit:
feature(SearchRadioComponentForm.SearchRadioComponentsButton_Click): добавлена проверка обработчика на null перед вызовом события SearchFinished

- Вызов события SearchFinished(...) переписан как SearchFinished?.Invoke(...)

Commit:
feature(SearchRadioComponentForm): удалена инициализация списка RadioComponents

Commit:
feature(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): проверка filteredIndexToRadioComponentMap.Count() == 0 заменена на filteredIndexToRadioComponentMap.Any()

Commit:
feature(Lab1View): добавлен класс с методами расширения для фильтрации радиокомпонентов WhereExtension

Commit:
feature(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): метод переписан через функции класса WhereExtension

fix(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): неверно выполнялась фильтрация

- добавлен оператор ! перед проверкой filteredIndexToRadioComponentMap.Any()

Commit:
feature(WhereExtension): добавлен метод ToIndexToRadioComponentMap

- ToIndexToRadioComponentMap возвращает перечислитель "индекс-радиокомпонент" на основе списка радиокомпонентов

Commit:
feature(WhereExtension): добавлен метод GetIndices

- метод GetIndices возвращает индексы перечислителя проиндексированных радиокомпонентов

Commit:
feature(SearchRadioComponentForm.GetFilteredRadioComponentsIndices): метод переписан с использованием методов WhereExtension.ToIndexToRadioComponentMap и WhereExtension.GetIndices

Commit:
style(SearchRadioComponentForm.cs): удалены неиспользуемые директивы using

Commit:
style(MainForm): отформатированы строки инициализации RadioComponents

Commit:
style(SearchRadioComponentForm): отформатированы строки объявления RadioComponents

Commit:
style(Lab1View): все формы перенесены в папку View

Commit:
style(Lab1View): все EventArgs классы перенесены в папку EventArgsClasses

Commit:
style(RadioComponentCreatedEventArgs.cs): удалена директива TEST

Commit:
style(SetRadioComponentSaveOptionForm.OkButton_Click): выбор опции сохранения переписан через тернарный оператор

Commit:
style(SetRadioComponentLoadOptionForm.OkButton_Click): выбор опции загрузки переписан через тернарный оператор

TemZZZ committed:
Merging: develop <-- console-loader-develop

Слияние ветви console-loader-develop (доработка ConsoleLoader) в ветвью develop

Commit:
reafactr(ConsoleLoader.cs): ComponentBase переименован в RadioComponentBase

Commit:
reafactr(Main.cs): ComponentBase переименован в RadioComponentBase

Commit:
feature(ConsoleLoader): добавлена ссылка на проект

ВАЖНОЕ ИЗМЕНЕНИЕ: добавлена ссылка на проект Lab1Model_DotNetFramework472\Lab1Model\Lab1Model\Lab1Model.csproj

Commit:
refactor(решение RadioComponentsCollection): переименовано из решения Lab1

Commit:
refactor(проект Model): переименован из проекта Lab1Model

Commit:
refactor(проект View): переименован из проекта Lab1View

Commit:
feractor(решение RadioComponentsCollection): изменена структура решения

- в проекте ConsoleLoader удалена ссылка на проект Model
- удалена папка Lab1Model с проектом Lab1Model под .NET Core из корневого каталога решения
- проект Model перенесен из папки Lab1Model_DotNetFramework472\Lab1Model в папку Model в корневом каталоге решения
- папка Lab1View с проектом View переименована в View

Commit:
refactor(проект View): папка с формами View переименована в Forms

Commit:
feature(ConsoleLoader): добавлена ссылка на проект Model

Commit:
feature(View): добавлена ссылка на проект Model

Commit:
refactor(Model): пространство имен Lab1Model переименовано в Model

Commit:
reafactor(View): пространство имен Lab1View переименовано в View

Commit:
style(проект View, Program.cs): удалены неиспользуемые директивы using

Commit:
feature(ConsoleLoader.csproj): установлено пространство имен по умолчанию ConsoleLoaderModel

Commit:
reafctor(Model.csproj): пространство имен по умолчанию и имя сборки изменены в Lab1Model на Model

Commit:
refactor(PositiveDoubleTextBoxLib.csproj): пространство имен по умолчанию и имя сборки переименованы из PositiveDoubleTextBox в PositiveDoubleTextBoxLib

Commit:
Revert "refactor(PositiveDoubleTextBoxLib.csproj): пространство имен по умолчанию и имя сборки переименованы из PositiveDoubleTextBox в PositiveDoubleTextBoxLib"

This reverts commit c1daf5d659d45eca649a1be2de9bfafb279a7305.

Commit:
refactor(PositiveDoubleTextBoxLib.csproj): пространство имен по умолчанию и имя сборки переименованы из PositiveDoubleTextBox в PositiveDoubleTextBoxLib

Commit:
refactor(View.csproj): пространство имен по умолчанию и имя сборки переименованы из Lab1View в View

Commit:
refactor(решение RadioComponentsCollection): переименована папка с проектом PositiveDoubleTextBoxLib

- удалена ссылка на проект PositiveDoubleTextBoxLib из проекта View
- удален проект PositiveDoubleTextBoxLib из решения
- папка PositiveDoubleTextBox переименована в PositiveDoubleTextBoxLib
- добавлен проект PositiveDoubleTextBoxLib из папки PositiveDoubleTextBoxLib
- добавлена ссылка на проект PositiveDoubleTextBoxLib в проекте View

Commit:
refactor(Model): в сведениях о сборке изменены название и продукт с Lab1Model на Model

Commit:
refactor(PositiveDoubleTextBoxLib): в сведениях о сборке изменены название и продукт с PositiveDoubleTextBox на PositiveDoubleTextBoxLib

Commit:
refactor(решение RadioComponentsCollection): переименована папка с проектом RegexTextBoxLib

- удалена ссылка на проект RegexTextBoxLib из проекта View
- удалена ссылка на проект RegexTextBoxLib из проекта PositiveDoubleTextBoxLib
- удален проект RegexTextBoxLib из решения
- папка RegularExpressionTextBoxLib переименована в RegexTextBoxLib
- добавлен проект RegexTextBoxLib из папки RegexTextBoxLib
- добавлена ссылка на проект RegexTextBoxLib в проекте View
- добавлена ссылка на проект RegexTextBoxLib в проекте PositiveDoubleTextBoxLib

Commit:
refactor(RegexTextBoxLib.csproj): пространство имен по умолчанию и имя сборки переименованы из RegularExpressionTextBoxLib в RegexTextBoxLib

Commit:
refactor(RegexTextBoxLib): в сведениях о сборке изменены название и продукт с RegularExpressionTextBoxLib на RegexTextBoxLib

Commit:
refactor(View): в сведениях о сборке изменены название и продукт с Lab1View на View

Commit:
feature(решение RadioComponentsCollection): добавлен проект RegexControlsSDK

- в папку проекта скопированы файлы PositiveDoubleTextBox.cs и PositiveDoubleTextBox.Designer.cs из проекта PositiveDoubleTextBoxLib
- в папку проекта скопированы файлы RegexTextBox.cs и RegexTextBox.Designer.cs из проекта RegexTextBoxLib
- в проект добавлены существующие (в папке поекта) элементы RegexTextBox и PositiveDoubleTextBox

Commit:
refactor(проект RegexControlsSDK, RegexTextBox.cs): пространство имен RegexTextBoxLib переименовано в RegexControlsSDK

Commit:
refactor(проект RegexControlsSDK, PositiveDoubleTextBox.cs): пространство имен PositiveDoubleTextBoxLib переименовано в RegexControlsSDK

Commit:
style(проект RegexControlsSDK, PositiveDoubleTextBox.cs): удалена неиспользуемая директива using RegexTextBoxLib

Commit:
refactor(проект RegexControlsSDK, RegexTextBox.Designer.cs): пространство имен RegexTextBoxLib переименовано в RegexControlsSDK

Commit:
refactor(проект RegexControlsSDK, PositiveDoubleTextBox.Designer.cs): пространство имен PositiveDoubleTextBoxLib переименовано в RegexControlsSDK

Commit:
feature(View): удалены ссылки на проекты RegexTextBoxLib и PositiveDoubleTextBoxLib

Commit:
feature(View): добавлена ссылка на проект RegexControlsSDK

Commit:
refactor(проект View, AddRadioComponentForm.cs): директива using PositiveDoubleTextBoxLib переименована в RegexControlsSDK

Commit:
refactor(проект View, Forms/AddRadioComponentForm.Designer.cs): PositiveDoubleTextBoxLib переименована в RegexControlsSDK

Commit:
refactor(проект View, Forms/SearchRadioComponentForm.Designer.cs): PositiveDoubleTextBoxLib переименована в RegexControlsSDK

Commit:
refactor(проект View, Forms/MainForm.cs): директива using PositiveDoubleTextBoxLib переименована в RegexControlsSDK

Commit:
refactor(проект View, Forms/MainForm.Designer.cs): PositiveDoubleTextBoxLib переименована в RegexControlsSDK

Commit:
feature(решение RadioComponentsCollection): удалены проекты RegexTextBoxLib и PositiveDoubleTextBoxLib

- удалена папка проекта RegexTextBoxLib из папки решения
- удалена папка проекта PositiveDoubleTextBoxLib из папки решения

Commit:
feature(решение RadioComponentsCollection): добавлен проект UnitTests

- в проект добавлена ссылка на nunit.framework
- в проект добавлена ссылка на проект Model
- в проект добавлена папка Model

Commit:
feature(UnitTests): добавлен класс ResistorTest

- подключено пространство имен NUnit.Framework
- класс помечен атрибутом [TestFixture] как тестовый

Commit:
feature(UnitTests): установлен пакет NUnit3TestAdapter

TemZZZ committed:
develop <-- regex-controls-sdk-develop

refactor(RadioComponentsCollection): элементы WinForm с проверкой на регулярные выражения вынесены в проект RegexControlsSDK

Commit:
feature(RadioComponentsCollection): слияние ветви regex-controls-sdk-develop в develop
